collect_app/src/main/java/org/odk/collect/android/logic/HierarchyElement.java

```diff
@@ -18,97 +18,115 @@
+        color = builder.color;
     }
 
-
-    public String getSecondaryText() {
-        return secondaryText;
+    public Drawable getIcon() {
+        return icon;
     }
 
-
-    public void setPrimaryText(String text) {
-        primaryText = text;
+    public ArrayList<HierarchyElement> getItemsAtLevel() {
+        return itemsAtLevel;
     }
 
-
-    public void setSecondaryText(String text) {
-        secondaryText = text;
+    public HierarchyElement getParent() {
+        return parent;
     }
 
-
-    public void setIcon(Drawable icon) {
-        this.icon = icon;
+    public String getPrimaryText() {
+        return primaryText;
     }
 
-
-    public Drawable getIcon() {
-        return icon;
+    public String getSecondaryText() {
+        return secondaryText;
     }
 
-
     public FormIndex getFormIndex() {
         return formIndex;
     }
 
-
     public int getType() {
         return type;
     }
 
-
-    public void setType(int newType) {
-        type = newType;
-    }
-
-
-    public ArrayList<HierarchyElement> getChildren() {
-        return children;
-    }
-
-
-    public void addChild(HierarchyElement h) {
-        children.add(h);
-    }
-
-
-    public void setChildren(ArrayList<HierarchyElement> children) {
-        this.children = children;
-    }
-
-
-    public void setColor(int color) {
-        this.color = color;
-    }
-
-
     public int getColor() {
         return color;
     }
 
+    public static class HierarchyElementBuilder {
+        private int type;
+        private FormIndex formIndex;
+        private ArrayList<HierarchyElement> itemsAtLevel;
+        private HierarchyElement parent;
+        private String primaryText = "";
+        private String secondaryText = "";
+        private Drawable icon;
+        private int color;
+
+        public HierarchyElementBuilder setParent(HierarchyElement parent) {
+            this.parent = parent;
+            return this;
+        }
+
+        public HierarchyElementBuilder setType(int newType) {
+            type = newType;
+            return this;
+        }
+
+        public HierarchyElementBuilder setFormIndex(FormIndex formIndex) {
+            this.formIndex = formIndex;
+            return this;
+        }
+
+        public HierarchyElementBuilder setItemsAtLevel(ArrayList<HierarchyElement> itemsAtLevel) {
+            this.itemsAtLevel = itemsAtLevel;
+            return this;
+        }
+
+        public HierarchyElementBuilder setPrimaryText(String primaryText) {
+            this.primaryText = primaryText;
+            return this;
+        }
+
+        public HierarchyElementBuilder setSecondaryText(String secondaryText) {
+            this.secondaryText = secondaryText;
+            return this;
+        }
+
+        public HierarchyElementBuilder setIcon(Drawable icon) {
+            this.icon = icon;
+            return this;
+        }
+
+        public HierarchyElementBuilder setColor(int color) {
+            this.color = color;
```

Comment:
What kind of color does this builder require? Will this be a color that's already resolved or will it be a resource? To make this alot clearer you could either name the variable `colorInt` or `colorRes`. You could also use the annotations `@ColorInt` or  `@ColorRes` so no confusion takes place.